ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/MessageScrubber.java

```diff
@@ -327,4 +358,40 @@ private static boolean bug1733118Affected(String namespace, String docType, Obje
             .matches("0[.]([0-9]|1[0-6])[.].*"); // < 0.17
   }
 
+  // See bug 1751753 for explanation of context.
+  private static void processForBug1751753(ObjectNode json) {
+    // Sanitize keys in the SEARCH_COUNTS histogram.
+    JsonNode searchCounts = json.path("payload").path("keyedHistograms").path("SEARCH_COUNTS");
+    if (searchCounts.isObject()) {
+      processKeysForBug1751753((ObjectNode) searchCounts);
+    }
+
+    // Sanitize keys in browser.search.content.* keyed scalars.
+    json.path("payload").path("processes").path("parent").path("keyedScalars") //
+        .fields().forEachRemaining(entry -> {
+          if (entry.getKey().startsWith("browser.search.content.") && entry.getValue().isObject()) {
+            processKeysForBug1751753((ObjectNode) entry.getValue());
+          }
+        });
+  }
+
+  private static void processKeysForBug1751753(ObjectNode searchNode) {
+    Lists.newArrayList(searchNode.fieldNames()).forEach(name -> {
+      if (name.contains(".in-content")) {
+        // This is an in-content search with format:
+        // <provider>.in-content:[sap|sap-follow-on|organic]:[<code>|none]
+        final String code = StringUtils.substringAfterLast(name, ":");
```

Comment:
I'm realizing this piece is a bit fragile. If the query param content contains a ":" character, then this could end up stripping only part of the user-provided value. I may need to use regexes and have different logic for the histogram and scalar cases.